lib/ASTGen/Sources/ASTGen/Misc.swift

```diff
@@ -26,4 +26,46 @@ extension ASTGenVisitor {
   public func visit(_ node: ArrayElementSyntax) -> ASTNode {
     visit(node.expression)
   }
+
+  /// Form a source location at the given absolute position in the current
+  /// file.
+  public func bridgedSourceLoc(at position: AbsolutePosition) -> BridgedSourceLoc {
+    return createBridgedSourceLoc(at: position, in: base)
+  }
+
+  /// Form a source location at the given node's position in the current file.
+  public func bridgedSourceLoc<Node: SyntaxProtocol>(for node: Node?) -> BridgedSourceLoc {
+    guard let node = node else {
+      return .none
+    }
+    return createBridgedSourceLoc(at: node.position, in: base)
+  }
+}
+
+extension BridgedSourceLoc {
```

Comment:
You could also make

```swift
extension BridgedSourceLoc: ExpressibleByNilLiteral {
  init(nilLiteral: ()) {
    self.init(raw: nil)
  }
}
```

Reply:
I... don't know how I feel about this 🤔 Any thoughts @rintaro / @DougGregor?

Reply:
FWIW I went ahead and did this. It's a little weird because for the optional case you now have `foo = someOptional ?? nil` 😅. But it overall does look a little better... 🤷